server/shadow/shadow_server.c

```diff
@@ -510,16 +536,50 @@ int shadow_server_start(rdpShadowServer* server)
 		return -1;
 	}
 
-	if (!server->ipcSocket)
-		status = server->listener->Open(server->listener, NULL, (UINT16)server->port);
-	else
-		status = server->listener->OpenLocal(server->listener, server->ipcSocket);
+	/* Bind to TCP port if nothing is specified, otherwise bind to everything specified */
+	ipc = server->ipcSocket && (strncmp(bind_address, server->ipcSocket,
+	                                    strnlen(bind_address, sizeof(bind_address))) != 0);
+	if (!ipc)
+	{
+		char* address =
+		    (server->ipcSocket)
+		        ? _strdup(server->ipcSocket + strnlen(bind_address, sizeof(bind_address)))
+		        : NULL;
+		char* modaddr = address;
 
-	if (!status)
+		if (modaddr)
+		{
+			if (modaddr[0] == '[')
```

Comment:
So this requires addresses to be in square brackets, including IPv4?

Reply:
Maybe this parsing code shouldn't be here but in `freerdp_tcp_resolve_host`?

Reply:
not really, as that is command line escape (we already use `:` for something else).
libfreerdp expects a valid DNS name or ip address